sys/net/gnrc/link_layer/lwmac/lwmac_internal.c

```diff
@@ -0,0 +1,538 @@
+/*
+ * Copyright (C) 2015 Daniel Krebs
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ * @ingroup     net_lwmac
+ * @file
+ * @brief       Internal functions of LWMAC
+ *
+ * @author      Daniel Krebs <github@daniel-krebs.net>
+ * @}
+ */
+
+#include <stdbool.h>
+#include <periph/rtt.h>
+#include <net/gnrc.h>
+#include <net/gnrc/lwmac/lwmac.h>
+#include <net/gnrc/lwmac/packet_queue.h>
+
+#include "include/lwmac_internal.h"
+#include "include/lwmac_types.h"
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+/******************************************************************************/
+
+int _get_dest_address(gnrc_pktsnip_t* pkt, uint8_t* pointer_to_addr[])
+{
+    int res;
+    gnrc_netif_hdr_t* netif_hdr;
+
+    if(!pkt)
+        return -ENODEV;
+
+    netif_hdr = (gnrc_netif_hdr_t*) pkt->data;
+    if( (res = netif_hdr->dst_l2addr_len) <= 0)
+        return -ENOENT;
+
+    *pointer_to_addr = gnrc_netif_hdr_get_dst_addr(netif_hdr);
+    return res;
+}
+
+/******************************************************************************/
+
+/* Find a payload based on it's protocol type */
+void* _gnrc_pktbuf_find(gnrc_pktsnip_t* pkt, gnrc_nettype_t type)
+{
+    while(pkt != NULL)
+    {
+        if(pkt->type == type) {
+            return pkt->data;
+        }
+        pkt = pkt->next;
+    }
+    return NULL;
+}
+
+/******************************************************************************/
+
+int _find_neighbour(lwmac_t* lwmac, uint8_t* dst_addr, int addr_len)
+{
+    lwmac_tx_neighbour_t* neighbours = lwmac->tx.neighbours;
+
+    for(int i = 0; i < LWMAC_NEIGHBOUR_COUNT; i++) {
```

Comment:
shouldn't this be ```for(int i = 0; i <= LWMAC_NEIGHBOUR_COUNT; i++)```   ?
since ```lwmac->tx.neighbours ``` has ```LWMAC_NEIGHBOUR_COUNT + 1 ``` members.